### PR TITLE
Remove usages of deprecated Apache Commons HttpClient 3.x library

### PR DESCRIPTION
**Pull request by basil**
### Problem

`ProxyConfiguration#doValidateProxy` continues to use the deprecated [Apache Commons HttpClient 3.x](https://hc.apache.org/httpclient-legacy/) library. Per its home page:

> The Commons HttpClient project is now end of life, and is no longer being developed. 

Continuing to rely on end-of-life (EOL) software does not seem sustainable.

### Solution

Now that we're on Java 11, we have a native HTTP client available, so just use that.

### Implementation

The port was straightforward, except for the fact that the original code attempted to detect `NTCredentials` and use NTLM authentication for such a use case, in contrast to the non-validation (i.e., production) code path which doesn't make this attempt. I simply removed any custom handling for NTLM, which makes the validation code consistent with the non-validation (i.e., production) code. In other words, the `Authenticator` used for validation (in `ProxyConfiguration#createAuthenticator`) is now symmetric with the `Authenticator` used in production (in `ProxyConfiguration#newAuthenticator`).

### Testing Done

I used https://hub.docker.com/r/lifenz/squid-docker-simple-auth to test this change. I was not able to validate the proxy server when the wrong port was entered or when the username or password were incorrect. I was able to validate the proxy server when the username and password were correct.

### Future Work

The production code path uses the older `URL#openConnection` API rather than the newer `java.net.http.HttpClient` API. This could be cleaned up in the future, probably resulting in further deduplication of code between the validation path and the production path. I am explicitly not tackling this cleanup in this PR.

### Proposed changelog entries

N/A

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change) and are in the imperative mood. [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  - Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadoc, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")` if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content-Security-Policy directives (see [documentation on jenkins.io](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`:

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are accurate, human-readable, and in the imperative mood
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/7094"></a>

### CODE REVIEW

**Review by NotMyFault — APPROVED** [review]

**Review by timja — APPROVED** [review]

**Comment by timja** [comment]
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
